src/cpp/src/continuous_batching/pipeline_base.cpp

```diff
@@ -183,33 +183,42 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         const auto& prompt = prompts[0];
         auto start_get_inputs_embeds = std::chrono::steady_clock::now();
 
-        encoded_images = m_inputs_embedder->encode_images(images_vector[0]);
+        encoded_images = m_inputs_embedder->encode_images(images_vector.size() > 0 ? images_vector[0] : std::vector<ov::Tensor>{});
         m_history_images.insert(m_history_images.end(), encoded_images.begin(), encoded_images.end());
 
-        encoded_videos = m_inputs_embedder->encode_videos(videos_vector[0]);
+        encoded_videos = m_inputs_embedder->encode_videos(videos_vector.size() > 0 ? videos_vector[0] : std::vector<ov::Tensor>{});
         m_history_videos.insert(m_history_videos.end(), encoded_videos.begin(), encoded_videos.end());
 
         auto [unified_prompt, image_sequence, video_sequence] = m_inputs_embedder->normalize_prompt(prompt, m_image_id, m_video_id, encoded_images, encoded_videos);
 
         m_history.push_back({{"role", "user"}, {"content", unified_prompt}});
         m_history_image_ids.insert(m_history_image_ids.end(), image_sequence.begin(), image_sequence.end());
         m_history_video_ids.insert(m_history_video_ids.end(), video_sequence.begin(), video_sequence.end());
+        m_history_vision_count.emplace_back(std::make_pair(video_sequence.size(), image_sequence.size()));
```

Comment:
Just curious why video count was chosen to be the first in pair? In other logic and in methods arguments (e.g. `get_inputs_embeds_with_token_type_ids`, `add_request`, `normalize_prompt`, `get_inputs_embeds`, etc.) images come first.

Reply:
For model QWen-VL, it need to calc position id which depends on video and image position in normalized prompt.

In one time chat, video should be putted to ahead of images, but in multiple chats with history, we also keep this relationshape.

for example:
one time chat:
xxxxxx<vision_start><video_pad><vision_end><vision_start><video_pad><vision_end>xxxxxxxxxxxxxxxx

multiple loop chat:

**loop_1**_x<vision_start><video_pad><vision_end><vision_start><video_pad><vision_end>x**loop_1**_end_xxx**loop_2**_x<vision_start><video_pad><vision_end><vision_start><video_pad><vision_end>x**loop_2**_end